workflow/artifacts/common/common.go

```diff
@@ -0,0 +1,14 @@
+package common
+
+import "github.com/argoproj/argo/v3/pkg/apis/workflow/v1alpha1"
+
+// ArtifactDriver is the interface for loading and saving of artifacts
+type ArtifactDriver interface {
+	// Load accepts an artifact source URL and places it at specified path
+	Load(inputArtifact *v1alpha1.Artifact, path string) error
```

Comment:
is there an expectation that the implementation modifies the input argument inputArtifact? if not, maybe it should be non-pointer to design-out bugs

Reply:
This is existing code, if we want to do this it should be in a separate PR